include/librealsense2/dds/dds-device-watcher.cpp

```diff
@@ -121,17 +124,20 @@ void dds_device_watcher::init()
 
     if( ! _listener )
         _participant->create_listener( &_listener )->on_writer_removed( [this]( dds::dds_guid guid, char const * ) {
-            std::lock_guard< std::mutex > lock( _devices_mutex );
-            auto it = _dds_devices.find( guid );
-            if( it != _dds_devices.end() )
+            std::shared_ptr< dds_device > device;
             {
-                auto device = it->second;
-                auto serial_number = device->device_info().serial;
-                LOG_DEBUG( "DDS device s/n " << serial_number << " removed from domain" );
-                if( _on_device_removed )
-                    _on_device_removed( device );
-                _dds_devices.erase( it );
+                std::lock_guard< std::mutex > lock( _devices_mutex );
+                auto it = _dds_devices.find( guid );
+                if( it != _dds_devices.end() )
+                {
+                    device = it->second;
+                    auto serial_number = device->device_info().serial;
+                    //LOG_DEBUG( "DDS device s/n " << serial_number << " removed from domain" );
```

Comment:
Printed elsewhere? Expected to be printed on the callback? If not needed delete

Reply:
Deleted, but for next PR.